### PR TITLE
Implement attr()/lattr() in MappedFile and DirectoryFilesystemEntry

### PR DESCRIPTION
**Pull request by pyrco**
(DIS-1795)

### CODE REVIEW

File: dissect/target/filesystem.py [hunk 1]
```diff
@@ -908,6 +908,16 @@ def stat(self) -> fsutil.stat_result:
     def lstat(self) -> fsutil.stat_result:
         return fsutil.stat_result.copy(os.lstat(self.entry))
 
+    def attr(self, follow_symlinks=True) -> list:
```
[the remark this answers is not in the document]
Comment (pyrco) [reply dissect/target/filesystem.py:911]: I don't know if it is a `'list[str]`, from what I've seen its more a `list[tuple[str]]` or `list[list[str]]`. But as it is not really clear I'd rather keep it at `list`.

[the remark this answers is not in the document]
Comment (pyrco) [reply dissect/target/filesystem.py]: I played around a bit, `os.listxattr` indeed gives back only strings, which are the names of the different attributes. This is not very useful in itself as you'd also want the values. I'm going to rework these functions to give a back a key/value dict.

**Review by cecinestpasunepipe — DISMISSED** [review]
LGTM

**Comment by Schamper** [comment]
Just a mental note that we should think of a standardized way to represent attr/xattr's. This currently uses a dictionary of str/bytes, but `extfs` uses a list of `XAttr` objects (from the dissect.extfs project). Perhaps create a separate ticket for that (if there's not one already)?

**Comment by pyrco** [comment]
Indeed, and ntfs uses `AttrributeMap`s. Furthemore these Windows attributes, on a normal filesystem, are retrieved using `os.stat()` which gives back another kind of format.

Also the classic Linux fs attributes (retrieved using `lsattr` etc.) have no easy way to retrieve using Python (except for using `ioctl()`).

**Pull request merged**